lib/more_core_extensions/core_ext/array/tableize.rb

```diff
@@ -118,7 +119,8 @@ def format_row(row, widths, justifications)
       end
 
       def format_field(field, width, justification)
-        field = field.to_s.gsub(/\n|\r/, '').slice(0, width)
+        field = field.to_s.gsub(/\n|\r/, '')
+        field = field.slice(0, width) unless field.include?("\e")
```

Comment:
We have to clip the width to _something_ because otherwise any long field will go over the max_width. I think the following will work:

```ruby
field = field.to_s.gsub(/\n|\r/, '').slice(0, width)
filed << "\e[0m" if field.include?("\e")
```

This can be tested by duplicating the "with really long column value" and the "'with really long column value and :max_width option'' and having a color variant of those.  You may also want test variants where the max_width comes after an escape sequence, before an escape sequence, and directly inside an escape sequence.

Reply:
So, yeah, I wrote this off the cuff, and it has issues, but the following seems to work...it's a bit brute forcey, but seems ok to me:

```ruby
      def format_field(field, width, justification)
        field = field.to_s.gsub(/\n|\r/, '')

        if field.include?("\e")
          while strip_ansi(field).size > width
            field.chop!
          end
          field << ANSI_RESET if field.include?("\e") && !field.end_with?(ANSI_RESET)
        else
          field = field.slice(0, width)
        end

        "%0#{justification}#{width}s" % field
      end

      def strip_ansi(field)
        field.gsub(ANSI_ESCAPE_SEQUENCE, '')
      end
```